### PR TITLE
[core][autoscaler] Update autoscaler proto for default enum value 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
When the `RUNNING` enum is 0, it doesn't show up when printing on the product side (which could be confusing) 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
cc @hanming-lu @brucez-anyscale

**Review by hanming-lu — APPROVED** [review]

File: src/ray/protobuf/experimental/autoscaler.proto [hunk 1]
```diff
@@ -72,7 +72,7 @@ message GangResourceRequest {
   repeated ResourceRequest requests = 1;
   // TODO(rickyx):
   // Strategy of the placement group, e.g. STRICT_PACK.
-  string pg_strategy = 2;
+  string placement_group_strategy = 2;
```
Comment (scv119) [inline src/ray/protobuf/experimental/autoscaler.proto:75]: can we call it placement strategy?

Comment (scv119) [reply src/ray/protobuf/experimental/autoscaler.proto:75]: hmm, actually why does placement group info leaked here?

File: src/ray/protobuf/experimental/autoscaler.proto [hunk 2]
```diff
@@ -85,12 +85,14 @@ message ClusterResourceConstraint {
 
 // Node status for a ray node.
 enum NodeStatus {
+  // Unknown state.
+  UNKNOWN = 0;
```
Comment (brucez-anyscale) [inline src/ray/protobuf/experimental/autoscaler.proto:89]: usually 0 is UNSPECIFIED
UNKNOWN can be 4 if you need

File: src/ray/protobuf/experimental/autoscaler.proto [hunk 3]
```diff
@@ -191,6 +193,20 @@ message PendingInstanceRequest {
   int32 count = 3;
 }
 
+message PendingInstance {
```
Comment (rickyyx) [inline src/ray/protobuf/experimental/autoscaler.proto:196]: I think this doesn't need to be populated in the first version.

File: src/ray/protobuf/experimental/autoscaler.proto [hunk 4]
```diff
@@ -70,9 +70,9 @@ message ResourceRequestByCount {
 message GangResourceRequest {
   // a map from bundles to the number of bundles requested.
   repeated ResourceRequest requests = 1;
-  // TODO(rickyx):
-  // Strategy of the placement group, e.g. STRICT_PACK.
-  string pg_strategy = 2;
+  // Metadata associated with the request for observability,
+  // e.g. placement group's strategy.
+  string details = 2;
```
Comment (hanming-lu) [inline src/ray/protobuf/experimental/autoscaler.proto:75]: Does autoscaler need to have any logic based on details?

Comment (rickyyx) [reply src/ray/protobuf/experimental/autoscaler.proto:75]: No - the purpose is to not have autoscaler depend on this. This is mainly for `ray status` - we might remove this in the futrue if we use 2 different endpoints for autoscaler and ray status.

File: src/ray/protobuf/experimental/autoscaler.proto [hunk 5]
```diff
@@ -191,6 +193,20 @@ message PendingInstanceRequest {
...
+  string instance_type_name = 1;
+  // The node type name, e.g. ray-head-node, matching `available_node_types`
+  // in the autoscaler config. See `ray/autoscaler/ray-schema.json`
+  string ray_node_type_name = 2;
+  // The instance id.
+  string instance_id = 3;
+  // The instance ip address.
+  string ip_address = 4;
+  // Detail information about the instance.
+  string details = 5;
```
Comment (hanming-lu) [inline src/ray/protobuf/experimental/autoscaler.proto:207]: What does details include?

Comment (rickyyx) [reply src/ray/protobuf/experimental/autoscaler.proto:207]: maybe zone/detail state information. This doesn't have to be populated if there's none. OSS autoscaler with more info could choose to populate it through this.

**Review by scv119 — APPROVED** [review]

**Pull request merged**